addons/website_sale_loyalty/tests/test_website_sale_loyalty_delivery.py

```diff
@@ -99,3 +101,44 @@ def test_shop_sale_gift_card_keep_delivery(self):
         admin_user.partner_id.write({'property_delivery_carrier_id': self.normal_delivery.id})
 
         self.start_tour("/", 'shop_sale_loyalty_delivery', login='admin')
+
+    def test_shipping_discount(self):
+        self.env['product.product'].create({
+            'name': 'Plumbus',
+            'list_price': 100.0,
+            'website_published': True,
+        })
+        self.env['loyalty.program'].create({
+            'name': 'Buy 3 get free shipping up to 75$',
+            'program_type': 'promotion',
+            'applies_on': 'current',
+            'trigger': 'auto',
+            'rule_ids': [(0, 0, {
+                'minimum_amount': 300.0
+            })],
+            'reward_ids': [(0, 0, {
+                'reward_type': 'shipping',
+                'discount_max_amount': 75.0
+            })],
+        })
+        product_paid_delivery = self.env['product.product'].create({
+            'name': 'free shipping (Max 75$)',
+            'invoice_policy': 'order',
+            'type': 'service',
+        })
+        delivery_with_rule = self.env['delivery.carrier'].create({
```

Comment:
why do you want to use delivery on rule? The basic one with fixed price can do the job, no?

Reply:
I didn't write the tests and they are ok IMO. I'm just lending a hand on the hard eCommerce part.

The basic one would probably do the trick but I don't want to invest time on that.